### DISCUSSION TITLE
Building and debugging EdgeTX (Windows 10 (MSYS2))

### DISCUSSION THREAD

**Discussion by Ktr128** [EdgeTX/edgetx#1726] [Q&A]
Hello.
Tried several times to install according to the instructions:
https://github.com/EdgeTX/edgetx/wiki/Build-Instructions-under-Windows-10-%28MSYS2%29

The third time I installed it on a PC with a reinstalled operating system (64-bit Windows 10 version 21H1).
I did everything exactly in accordance with the instructions, the probability of error is minimal.
The error occurs at the stage - Second stage of MSYS2 configuration. I am attaching screenshots.
Can you please tell me how to fix these errors?

<a href="https://ibb.co/bK42PN8"><img src="https://i.ibb.co/fx3n2rP/1.png" alt="1" border="0"></a>

<a href="https://ibb.co/yhKt7f1"><img src="https://i.ibb.co/HNR63Fs/2.png" alt="2" border="0"></a>

**Comment by pfeerick** [comment]
The issue here is that MSYS2 only officially keeps the current versions of packages in their repository, and don't support installing older versions of packages. Pillow 7.2.0, a package used for generating images for part of the build process, only works with Python 3.5 - 3.8, and with a new install of MSYS2, you're running Python 3.9 (at time of writing). 

As a short term workaround (very short, since it is depending on how long the files will still be on the msys2 repo), you *could possibly* install python 3.8 manually - i.e. download and install the package and then running that stage2 script again.

```bash 
curl -LO http://repo.msys2.org/mingw/mingw64/mingw-w64-x86_64-python-3.8.9-2-any.pkg.tar.zst
pacman -U mingw-w64-x86_64-python-3.8.9-2-any.pkg.tar.zst
```

Alternately, you could instead alter the `setup_buildenv_msys2_py_requirements.txt` file after it's downloaded at step 3 (you'll need to run install script with the `--pause` parameter so it stops after running each step)... and change the pillow version to something like 8.3.2... keeping in mind it may result in variances in the graphics produced for the UI since it's not the version we use and test with. However, if you just want to build and test something, it will probably be fine. 

Unfortunately, the step that failed also installs `aqtinstall`, hence the "aqt: command not found" error.

**Comment by Ktr128** [comment]
Thanks for the information.
I will try to install with your recommendations.
While temporarily made on ubuntu, everything worked out))

https://www.rcgroups.com/forums/showpost.php?p=48596381&postcount=4274

<a href="https://ibb.co/4j3WcC9"><img src="https://i.ibb.co/4j3WcC9/Screenshot-20220128-210607-Gallery.jpg" alt="Screenshot-20220128-210607-Gallery" border="0"></a>

**Comment by Dist0rtion** [comment]
I am currently compiling the firmware with Ubuntu without problems, but since i use mostly Window is it possible to update the build instructions under Windows 10 (MSYS2) to run with Python 3.9 - 3.11?
Also having the latest Companion version (nightly) running on Windwos is quite handy

Many Thanks

**Comment by rotorman** [comment]
@Dist0rtion EdgeTX needs presently specific versions of pillow and clang (see [here](https://github.com/EdgeTX/edgetx/blob/main/tools/setup_buildenv_msys2_py_requirements.txt)) and these do not match with newer Python versions unfortunately.

**Comment by Dist0rtion** [comment]
I know... unfortunately
Or could maybe someone provide the old, already downloaded/installed version of MSYS2 with the old Python version and dependencies? Just put the folder (C:\MSYS2) into a ZIP/RAR archive... or something like that.
So that you only have to execute the last step/script

**Comment by rotorman** [comment]
Are you aware of web-browser based building of EdgeTX radio firmware - possibly you can use this as an alternative - it has all the correct versions: https://github.com/EdgeTX/edgetx/wiki/Building-radio-firmware-in-a-webbrowser-with-Gitpod

**Comment by Dist0rtion** [comment]
Yes, this is of course a practical and easy alternative, but it does not compile the Companion/Simulator (for Win). 
Or does that work with it as well?
Not that I insist on compiling the companion app, but it would be handy :)

**Comment by rotorman** [comment]
True, you cannot unfortunately (yet) build Windows binaries with EdgeTX Gitpod.

**Comment by Ktr128** [comment]
Unable to compile firmware v2.7.0-rc1 for nv14
In the process of compilation, an error occurs associated with an external module.
(/workspace/edgetx/radio/src/pulses/pulses.cpp:705:7: **error**: 'extmoduleSendBuffer' was not declared in this scope
  705 |       extmoduleSendBuffer(extmodulePulsesData.afhds3.getData(), )

I compile with the following options:
cmake -DPCB=NV14 -DDEFAULT_MODE=2 -DGVARS=YES -DPPM_UNIT=US -DHELI=NO -DLUA=YES -DINTERNAL_GPS=NO -DCMAKE_BUILD_TYPE=Release ../

Tell me please, what could be the problem?

**Reply by pfeerick** [reply]
Remove your CMakeCache.txt, clean the cache `make clean` and run cmake and make again. Same as https://github.com/EdgeTX/edgetx/issues/1738

**Reply by Ktr128** [reply]
Everything worked out. Thanks for the help!

**Comment by WillyZehnder** [comment]
On Linux the complete build of all parts of EdgeTX using Docker is running perfectly.
But because my main OS is Windows and the Cloudbuild doesn't offer the switches I need for my X9E (LUA_MIXER=Y, GHOST:BOOL=FALSE, ...) I tried to build the Companion+Simulator on Windows according to https://github.com/EdgeTX/edgetx/wiki/Build-Instructions-under-Windows-10-%28MSYS2%29

I know that Windows 10 is the necessary OS, but I gave Win 11 a try:
Microsoft Windows 11 Home
Version	10.0.26100 Build 26100

Unfortunatelly that doesn't work. I've got an error like this:
<img width="1620" height="1022" alt="Screenshot 2025-11-15 164416" src="https://github.com/user-attachments/assets/87f904f1-8612-4c1e-8d84-d23ee1870bdf" />

**Is there a chance to do that building on Windows 11** (in the future), because I guess most users will change to Win 11 after stop of support for Win 10?

**Reply by rotorman** [reply]
Have a look at: https://github.com/EdgeTX/edgetx/wiki/Build-Instructions-under-Windows

**Reply by WillyZehnder** [reply]
I had a look there before I asked here, but unfortunatelly the X9E is only supported 'till V2.11.x
`Note: The instructions on this page apply only for building V3.0.0-dev and later versions of EdgeTX.`
So I'm afraid that's no option for me.

I still need a solution to build a Windows Companion incl. Simulator for V2.x.y with free selectable options.

**Reply by WillyZehnder** [reply]
@rotorman At https://github.com/EdgeTX/edgetx/wiki/Build-Instructions-under-Windows there is a hint 
`Note: The instructions on this page apply only for building V3.0.0-dev and later versions of EdgeTX.`
but V3.x.y isn't existing. On the other hand I can see a 2.12 ?

So I'm insecure starting at which version that Build-Instructions will work correctly. 
Could you clarify that, please.

**Reply by rotorman** [reply]
Current main branch is v3-dev.

**Reply by WillyZehnder** [reply]
Thank you for the quick reply @rotorman 
I have learned that not only tags and branches are interesting - I should also have a look at the CMakeLists.txt

**Comment by WillyZehnder** [comment]
So I still try to use  https://github.com/EdgeTX/edgetx/wiki/Build-Instructions-under-Windows-10-%28MSYS2%29 on Windows 11.

After preparing the environment (manual unzip) successfully, I get an error at "Building Firmware Binary" that I can't find the root cause of.
[Error on Building firmware binary.txt](https://github.com/user-attachments/files/23569412/Error.on.Building.firmware.binary.txt)
**Has anybody an idea of the cause of that?**

Edit:
In the meantime I recognized that I made a mistake. My X9E is still supported - only the support for my QX7 is stopped with 2.11.

So it seems it's not really efficient to put effort to the deprecated 'Windows building with MSYS2' only for some old Transmitters.
I gave it a try, but there are many obstacles on that way: unzip->tar, Qt5->Qt6, Libfox,... - so, too much for my knowledge.

But if @rotorman could spend a last little time to adapt that building method to Windows 11 and ETX V2.11.3, I'm sure that many pilots with Jumper T-Lite, Jumper T-Pro, Jumper T12, RadioMaster T8 Pro, RadioMaster TX12, BetaFPV LiteRadio3 Pro, FrSky QX7, FrSky X9 Lite, FrSky X9D or FrSky X-Lite would be very happy to have a Companion and Simulator that's fitting to the capabilities of their transmitter firmware.
